src/helpers/Image/Image.js

```diff
@@ -0,0 +1,88 @@
+import { flattenToAppURL, isInternalURL } from '@plone/volto/helpers';
+import config from '@plone/volto/registry';
+
+/**
+ * Get src-set list from image
+ * @param {object | string} image - Image content object or url
+ * @param {object} options
+ * @param {number} options.maxSize - maximum size to render
+ * @param {boolean} options.useOriginal - whether to render original img
+ * @returns {object} image attributes
+ * @returns {string} image.src attributes.src
+ * @returns {string} image.srcset attributes.srcset
+ */
+export const getImageAttributes = (
+  image,
+  { imageField = 'image', maxSize = 10000, useOriginal = false } = {},
+) => {
+  const imageScales = config.settings.imageScales;
+  const minSize = Object.keys(imageScales).reduce((minSize, scale) => {
+    if (!minSize || imageScales[scale] < imageScales[minSize]) {
+      return scale;
+    }
+
+    return minSize;
+  }, null);
+
+  let attrs = {};
+  let imageType = 'external';
+  if (image['content-type'] === 'image/svg+xml') imageType = 'svg';
+  else if (
+    Object.prototype.toString.call(image) === '[object Object]' &&
+    image.scales &&
+    Object.keys(image.scales).length > 0
+  )
+    imageType = 'imageObject';
+  else if (typeof image === 'string' && isInternalURL(image))
+    imageType = 'internalUrl';
+
+  switch (imageType) {
+    case 'svg':
+      attrs.src = flattenToAppURL(image.download);
+      break;
+
+    // Scales object from Plone restapi
+    // ideal use of Plone images
+    case 'imageObject':
```

Comment:
I'm missing the use case more common in blocks. The one that you know the image field (coming from the catalog), and you want to render it, eg. an image in a teaser.

Granted, that is difficult to get right with the srcset, but might be you want it anyways (no srcset info).

Maybe, the possibility to fetch the info from the image object anyways?